### PR TITLE
Change the dataset used for unit test to fix issue on Linux

### PR DESCRIPTION
**Pull request by bpstark**
There is an issue with how some of the images were saved in the dataset
such that they were not being read properly on linux. As such accuracy
was dramatically reduced given the small size of the dataset to begin
with. Have corrected the images and made a new dataset such that this
issue should now be resolved.

### CODE REVIEW

**Comment by codemzs** [comment]
>             }

what about here?

---
Refers to: test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs:1375 in 800721d. [](commit_id = 800721d13fb7ca228fea022477e8f8aa1b622897, deletion_comment = False)

**Comment by codemzs** [comment]
>             }

and here?

---
Refers to: test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs:1279 in 800721d. [](commit_id = 800721d13fb7ca228fea022477e8f8aa1b622897, deletion_comment = False)

**Comment by codemzs** [comment]
>             // investigate this further.

can we now remove the special if condition for nix? and then get rid of this comment?

---
Refers to: test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs:1522 in 800721d. [](commit_id = 800721d13fb7ca228fea022477e8f8aa1b622897, deletion_comment = False)

**Comment by codemzs** [comment]
@bpstark This change does not fix the issue for lower accuracy completely. Please investigate why we are seeing lower accuracy on Linux.

File: test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs [hunk 1]
```diff
@@ -1857,6 +1781,12 @@ public void TensorFlowImageClassificationBadImages()
         {
             var files = Directory.GetFiles(folder, "*",
                 searchOption: SearchOption.AllDirectories);
+            /*
+             * This is only needed as Linux can produce files in a different 
+             * order than other OSs. As this is a test case we want to maintain
+             * consisten accuracy across all OSs, so we sort to remove this discrepency.
```
Comment (codemzs) [inline test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs]: >consisten [](start = 15, length = 9)

consistent

Comment (codemzs) [inline test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs]: >OSs [](start = 45, length = 3)

typo?

File: test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs [hunk 2]
```diff
@@ -1266,22 +1266,8 @@ public void TensorFlowImageClassificationDefault()
...
-            // build down. Accuracy values seen were 0.33, 0.66, 0.70+. The model
-            // seems to be unstable, there could be many reasons, will need to 
-            // investigate this further.
-            if (!(RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ||
-                (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))))
-            {
-                Assert.InRange(metrics.MicroAccuracy, 0.2, 1);
-                Assert.InRange(metrics.MacroAccuracy, 0.2, 1);
-            }
-            else
```
Comment (codemzs) [inline test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs]: also remove in AutoFitImageClassificationTrainTest

File: test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs [hunk 3]
```diff
@@ -1889,10 +1819,7 @@ public void TensorFlowImageClassificationBadImages()
         public static string DownloadImageSet(string imagesDownloadFolder)
         {
             string fileName = "flower_photos_tiny_set_for_unit_tests.zip";
-            string url = $"https://mlnetfilestorage.file.core.windows.net/imagesets" +
-                $"/flower_images/flower_photos_tiny_set_for_unit_tests.zip?st=2019" +
-                $"-08-29T00%3A07%3A21Z&se=2030-08-30T00%3A07%3A00Z&sp=rl&sv=2018" +
-                $"-03-28&sr=f&sig=N8HbLziTcT61kstprNLmn%2BDC0JoMrNwo6yRWb3hLLag%3D";
+            string url = "https://tlcresources.blob.core.windows.net/datasets/flower_photos_tiny_set_for_unit_test.zip";
```
Comment (justinormont) [inline test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs]: Please move to an aka.ms link pointing to the CDN. Recommend using the resource downloader class for your downloads.

Comment (justinormont) [inline test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs]: What kind of images are not loading? AutoML should detect this issue and notify the user. Just swapping out the images means users will still run into this issue.

Comment (bpstark) [reply test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs]: The issue ended up not being the images, but the ordering. I have changed the dataset, so we use a short link now, but the images were not bad, just had some odd metadata, that I removed as I thought that may have contributed to the issue initially, however that ended up not being the issue.

---
In reply to: [347016790](https://github.com/dotnet/machinelearning/pull/4475#discussion_r347016790) [](ancestors = 347016790)

Comment (justinormont) [reply test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs]: Can we fix the loader such that a user doesn't run into the same issue?

Does the fix address the underlying issue? Or does fixing the dataset just avoid the issue for our unit test?

Comment (justinormont) [reply test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs]: Ahh... the order of the dataset rows differed between Windows/Unix, which lead to different dataset splits, which lead to differing metrics.

If we haven't, we may want to modify our example code where we load a folder of images.

**Pull request merged**